pkg/bff/auth/cache/cache.go

```diff
@@ -0,0 +1,88 @@
+package cache
+
+import (
+	"errors"
+	"sync"
+
+	"github.com/trisacrypto/directory/pkg/bff/config"
+)
+
+type ResourceFetcher interface {
+	Get(key string) (interface{}, error)
+}
+
+// TTLCache is a time-to-live cache that relies on a key value store for persistence,
+// automatically refreshing entries when they expire and uses a FIFO eviction policy
+// to manage the cache size.
+type TTLCache struct {
+	sync.RWMutex
+	fetcher ResourceFetcher
+	store   KeyValueStore
+	conf    config.CacheConfig
+	keys    []string
+}
```

Comment:
I was thinking we eventually might want to use this for other Auth0 requests as well or utilize a different type of cache storage (e.g. disk) so this implementation is pretty generic.